Failure to compile with HELICS support

Initial Checklist prior to submitting the issue. Check the below boxes to the best of your knowledge:

- [x] I have not found an open issue describing the same problem.
- [x] I went through "CONTRIBUTING.MD" before deciding to submit the issue.
- [x] While building GridLAB-D, I followed all steps as detailed in GridLAB-D Wiki.

------------------------------------------------------------------------------------------------------------

## Please describe what was the cause of the issue?
I provided the "--with-helics=" argument to ./configure, pointing it to /home/username/.local/helics-2.3.0; when compiling. The compilation fails, citing two instances of `invalid operands of types 'std::size_t {aka long unsigned int}' and 'double' to binary 'operator<<'`, both in relation to HELICS messages.

## Please provide the steps to reproduce.
I was following [these instructions](https://github.com/GMLC-TDC/HELICS-Tutorial/blob/master/setup/README.md), specifically:
1. Install HELICS following [these instructions](https://helics.readthedocs.io/en/latest/installation/linux.html#building-helics-with-python-support). Was on https://github.com/GMLC-TDC/HELICS/commit/b6cf2eecb1958421d53044a182e442646a5b07d7 at time of writing.
2. Install HELICS-Runner using [these instructions](https://github.com/GMLC-TDC/helics-cli)
3. Clone this repo, checkout the develop branch, make a "build" directory, and from within it run
``` 
./configure --with-helics=/home/ericsilk/.local/helics-2.3.0 --enable-silent-rules "CFLAGS=-g -O0 -w" "CXXFLAGS=-g -O0 -w -std=c++14" "LDFLAGS=-g -O0 -w"
```
Was on commit ed817d27cc22bad8bc13184679882581d7276bfe at time of writing.

## Please provide the exact error/warnining/failure message printed on the console.
```
/bin/bash -c "source utilities/build_number . gldcore/build.h"
Updating gldcore/build.h: revision 18186 (ed817d27:develop:Mod)
/home/ericsilk/Documents/HELICS_Project/gridlab-d
make  all-am
make[1]: Entering directory '/home/ericsilk/Documents/HELICS_Project/gridlab-d'
  CXX      connection/connection_connection_la-helics_msg.lo
In file included from /usr/include/c++/7/bits/hashtable.h:35:0,
                 from /usr/include/c++/7/unordered_map:47,
                 from /home/ericsilk/.local/helics-2.3.0/include/helics/external/cereal/archives/../cereal.hpp:36,
                 from /home/ericsilk/.local/helics-2.3.0/include/helics/external/cereal/archives/portable_binary.hpp:32,
                 from /home/ericsilk/.local/helics-2.3.0/include/helics/application_api/ValueConverter_impl.hpp:20,
                 from /home/ericsilk/.local/helics-2.3.0/include/helics/application_api/ValueConverter.hpp:65,
                 from /home/ericsilk/.local/helics-2.3.0/include/helics/application_api/ValueFederate.hpp:11,
                 from /home/ericsilk/.local/helics-2.3.0/include/helics/application_api/CombinationFederate.hpp:10,
                 from connection/helics_msg.h:21,
                 from connection/helics_msg.cpp:16:
/usr/include/c++/7/bits/hashtable_policy.h: In member function ‘std::size_t std::__detail::_Power2_rehash_policy::_M_next_bkt(std::size_t)’:
/usr/include/c++/7/bits/hashtable_policy.h:563:40: error: invalid operands of types ‘std::size_t {aka long unsigned int}’ and ‘double’ to binary ‘operator<<’
       const auto __max_bkt = size_t(1) << (__max_width * __CHAR_BIT__ - 1);
                              ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Makefile:4574: recipe for target 'connection/connection_connection_la-helics_msg.lo' failed
make[1]: *** [connection/connection_connection_la-helics_msg.lo] Error 1
  CXX      connection/connection_connection_la-init.lo
In file included from /usr/include/c++/7/bits/hashtable.h:35:0,
                 from /usr/include/c++/7/unordered_map:47,
                 from /home/ericsilk/.local/helics-2.3.0/include/helics/external/cereal/archives/../cereal.hpp:36,
                 from /home/ericsilk/.local/helics-2.3.0/include/helics/external/cereal/archives/portable_binary.hpp:32,
                 from /home/ericsilk/.local/helics-2.3.0/include/helics/application_api/ValueConverter_impl.hpp:20,
                 from /home/ericsilk/.local/helics-2.3.0/include/helics/application_api/ValueConverter.hpp:65,
                 from /home/ericsilk/.local/helics-2.3.0/include/helics/application_api/ValueFederate.hpp:11,
                 from /home/ericsilk/.local/helics-2.3.0/include/helics/application_api/CombinationFederate.hpp:10,
                 from connection/helics_msg.h:21,
                 from connection/init.cpp:16:
/usr/include/c++/7/bits/hashtable_policy.h: In member function ‘std::size_t std::__detail::_Power2_rehash_policy::_M_next_bkt(std::size_t)’:
/usr/include/c++/7/bits/hashtable_policy.h:563:40: error: invalid operands of types ‘std::size_t {aka long unsigned int}’ and ‘double’ to binary ‘operator<<’
       const auto __max_bkt = size_t(1) << (__max_width * __CHAR_BIT__ - 1);
                              ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Makefile:4595: recipe for target 'connection/connection_connection_la-init.lo' failed
make[1]: *** [connection/connection_connection_la-init.lo] Error 1
/bin/bash -c "source utilities/build_number . gldcore/build.h"
Updating gldcore/build.h: revision 18186 (ed817d27:develop:Mod)
/home/ericsilk/Documents/HELICS_Project/gridlab-d
  CXX      gldcore/link/engine/gldcore_link_engine_glxengine_la-engine.lo
  CXXLD    gldcore/link/engine/glxengine.la
  CC       gldcore/gridlabd-version.o
  CXXLD    gridlabd
make[1]: Target 'all-am' not remade because of errors.
make[1]: Leaving directory '/home/ericsilk/Documents/HELICS_Project/gridlab-d'
Makefile:1612: recipe for target 'all' failed
make: *** [all] Error 2

```



## If possible, please provide any screenshots and other supporting information.

I'm on (K)Ubuntu 18.04. gcc version 7.4.0. 
Forcing the usage of clang (CC=clang CXX=clang++, version is 6.0.0-1ubuntu2), there is one similar error but also:
```
cannot pass object of non-trivial type 'std::__cxx11::string'
    (aka 'basic_string<char>') through variadic function; call will abort at runtime [-Wnon-pod-varargs]
        gl_verbose("helics_msg::init(): registering publishing endpoint: %s", gld_ep_pub->name);
                                                                                          ^
```
-------------------------------------------------------------------------------------------------------------

--- a/climate/climate.cpp
+++ b/climate/climate.cpp
@@ -19,12 +19,6 @@
 
 #include <string>
 #include "gridlabd.h"
-#ifdef max
-#undef max
-#endif
-#ifdef min
-#undef min
-#endif
 #include "climate.h"
 #include "timestamp.h"
 EXPORT_CREATE(climate)
@@ -1136,7 +1130,7 @@ void climate::update_cloud_pattern(TIMESTAMP delta_t) {
 	//double windspeed = windspeed_tmy2 * log(1000/roughness_length)/log(10/roughness_length);
 
 	//C. W. Hansen, J. S. Stein, and A. Ellis,
-	//“Simulation of One-Minute Power Output from Utility-Scale Photovoltaic Generation Systems,” SAND2011-5529, 2011.
+	//"Simulation of One-Minute Power Output from Utility-Scale Photovoltaic Generation Systems," SAND2011-5529, 2011.
 	// Shows distribution of wind speeds at cumulus cloud elevations based on weather balloon launches (pg. 18)
 	// A cursory inspection of the TMY3 measured wind speeds shows this is roughly consistent with histogram
 	// presented in the paper.

--- a/generators/diesel_dg.cpp
+++ b/generators/diesel_dg.cpp
@@ -3551,9 +3551,6 @@ STATUS diesel_dg::apply_dynamics(MAC_STATES *curr_time, MAC_STATES *curr_delta, 
 			x0=gov_gast_AT+gov_gast_KT*(gov_gast_AT-curr_time->gov_gast.x3);
 		}
 
-//		x0 = min(,gov_gast_AT+gov_gast_KT*(gov_gast_AT-curr_time->gov_gast.x3)); 
-//		LL+K*(LL-G1.machine_parameters.curr.gov.x3
-
 		//Update variables
 		curr_delta->gov_gast.x1 = (x0 - curr_time->gov_gast.x1)/gov_gast_T1;
 		curr_delta->gov_gast.x2 = (curr_time->gov_gast.x1 - curr_time->gov_gast.x2)/gov_gast_T2;

--- a/gldcore/debug.c
+++ b/gldcore/debug.c
@@ -541,19 +541,19 @@ Retry:
 			fprintf(stdout,"<Ctrl-C>\n");
 			goto Retry;
 		}
-		if (strncmp(cmd,"quit",max(1,strlen(cmd)))==0)
+		if (strncmp(cmd,"quit",fmax(1,strlen(cmd)))==0)
 			return DBG_QUIT;
-		else if (strncmp(cmd,"run",max(1,strlen(cmd)))==0 || (strlen(cmd)==0&&last==DBG_RUN))
+		else if (strncmp(cmd,"run",fmax(1,strlen(cmd)))==0 || (strlen(cmd)==0&&last==DBG_RUN))
 		{
 			output_debug("resuming simulation, Ctrl-C interrupts");
 			debug_active=0;
 			return last = DBG_RUN;
 		}
-		else if (strncmp(cmd,"next",max(1,strlen(cmd)))==0 || (strlen(cmd)==0&&last==DBG_NEXT))
+		else if (strncmp(cmd,"next",fmax(1,strlen(cmd)))==0 || (strlen(cmd)==0&&last==DBG_NEXT))
 		{
 			return last = DBG_NEXT;
 		}
-		else if (strncmp(cmd,"module",max(1,strlen(cmd)))==0)
+		else if (strncmp(cmd,"module",fmax(1,strlen(cmd)))==0)
 		{
 			char modname[128];
 			
@@ -569,7 +569,7 @@ Retry:
 				}
 			}
 		}
-		else if (strncmp(cmd,"namespace",max(2,strlen(cmd)))==0)
+		else if (strncmp(cmd,"namespace",fmax(2,strlen(cmd)))==0)
 		{
 			char space[1024];
 			if (sscanf(buffer,"%*s %s", space)==0)
@@ -580,7 +580,7 @@ Retry:
 			else if (!object_select_namespace(space))
 				output_debug("unable to select namespace '%s'", space);
 		}
-		else if (strncmp(cmd,"list",max(1,strlen(cmd)))==0)
+		else if (strncmp(cmd,"list",fmax(1,strlen(cmd)))==0)
 		{
 			char lclass[256]="";
 			OBJECT *obj = object_get_first();
@@ -595,7 +595,7 @@ Retry:
 					list_object(obj,pass);
 			}
 		}
-		else if (strncmp(cmd,"details",max(1,strlen(cmd)))==0)
+		else if (strncmp(cmd,"details",fmax(1,strlen(cmd)))==0)
 		{
 			char cmd[1024];
 			int n = sscanf(buffer,"%*s %[^\0]", cmd);
@@ -622,7 +622,7 @@ Retry:
 					and try again.
 				 */
 		}
-		else if (strncmp(cmd,"inactive",max(1,strlen(cmd)))==0)
+		else if (strncmp(cmd,"inactive",fmax(1,strlen(cmd)))==0)
 		{
 			char cmd[1024];
 			int n = sscanf(buffer,"%*s %[^\0]", cmd);
@@ -649,7 +649,7 @@ Retry:
 					and try again.
 				 */
 		}
-		else if (strncmp(cmd,"unnamed",max(1,strlen(cmd)))==0)
+		else if (strncmp(cmd,"unnamed",fmax(1,strlen(cmd)))==0)
 		{
 			char cmd[1024];
 			int n = sscanf(buffer,"%*s %[^\0]", cmd);
@@ -676,7 +676,7 @@ Retry:
 					and try again.
 				 */
 		}
-		else if (strncmp(cmd,"nsync",max(2,strlen(cmd)))==0)
+		else if (strncmp(cmd,"nsync",fmax(2,strlen(cmd)))==0)
 		{
 			char cmd[1024];
 			int n = sscanf(buffer,"%*s %[^\0]", cmd);
@@ -703,7 +703,7 @@ Retry:
 					and try again.
 				 */
 		}
-		else if (strncmp(cmd,"script",max(1,strlen(cmd)))==0)
+		else if (strncmp(cmd,"script",fmax(1,strlen(cmd)))==0)
 		{
 			char load_filename[_MAX_PATH+32]; /* Made it a little longer than max_path to be safe */
 			if(sscanf(buffer,"%*s %[^\x20]",load_filename) <=0) /* use \n since we are using fgets to load buffer */
@@ -725,7 +725,7 @@ Retry:
 			}
 			load_from_file = 1;
 		}
-		else if (strncmp(cmd,"system",max(2,strlen(cmd)))==0)
+		else if (strncmp(cmd,"system",fmax(2,strlen(cmd)))==0)
 		{
 			char cmd[1024];
 			if (sscanf(buffer,"%*s %[^\0]", cmd)==1)
@@ -742,7 +742,7 @@ Retry:
 				system("/bin/sh");
 #endif
 		}
-		else if (strncmp(cmd,"break",max(1,strlen(cmd)))==0)
+		else if (strncmp(cmd,"break",fmax(1,strlen(cmd)))==0)
 		{
 			char bptype[256]="";
 			char bpval[256]="";
@@ -871,9 +871,9 @@ Retry:
 			else if (strncmp(bptype,"pass",strlen(bptype))==0)
 			{	/* create pass breakpoint */
 				int pass;
-				if (strnicmp(bpval,"pretopdown",max(2,strlen(bpval)))==0) pass=PC_PRETOPDOWN;
+				if (strnicmp(bpval,"pretopdown",fmax(2,strlen(bpval)))==0) pass=PC_PRETOPDOWN;
 				else if (strnicmp(bpval,"bottomup",strlen(bpval))==0) pass=PC_BOTTOMUP;
-				else if (strnicmp(bpval,"posttopdown",max(2,strlen(bpval)))==0) pass=PC_POSTTOPDOWN;
+				else if (strnicmp(bpval,"posttopdown",fmax(2,strlen(bpval)))==0) pass=PC_POSTTOPDOWN;
 				else
 				{
 					output_error("undefined pass type for add breakpoint");
@@ -972,7 +972,7 @@ Retry:
 					Check the command syntax and try again.
 				 */
 		}
-		else if (strncmp(cmd,"watch",max(2,strlen(cmd)))==0)
+		else if (strncmp(cmd,"watch",fmax(2,strlen(cmd)))==0)
 		{
 			char wptype[256]="";
 			char wpval[256]="";
@@ -1083,7 +1083,7 @@ Retry:
 				Check the command syntax and try again.
 			 */
 		}
-		else if (strncmp(cmd,"where",max(1,strlen(cmd)))==0)
+		else if (strncmp(cmd,"where",fmax(1,strlen(cmd)))==0)
 		{
 			char ts[64];
 			output_debug("Global clock... %s (%" FMT_INT64 "d)", convert_from_timestamp(global_clock,ts,sizeof(ts))?ts:"(invalid)",global_clock);
@@ -1094,7 +1094,7 @@ Retry:
 			output_debug("Rank........... %d", index);
 			output_debug("Object......... %s",get_objname(obj));
 		}
-		else if (strncmp(cmd,"print",max(1,strlen(cmd)))==0)
+		else if (strncmp(cmd,"print",fmax(1,strlen(cmd)))==0)
 		{
 			char tmp[4096];
 			if (sscanf(buffer,"%*s %s",tmp)==0)
@@ -1128,7 +1128,7 @@ Retry:
 				 */
 			}
 		}
-		else if (strncmp(cmd,"globals",max(1,strlen(cmd)))==0)
+		else if (strncmp(cmd,"globals",fmax(1,strlen(cmd)))==0)
 		{
 			GLOBALVAR *var;
 			for (var=global_getnext(NULL); var!=NULL; var=global_getnext(var))
@@ -1140,7 +1140,7 @@ Retry:
 				output_message("%-32.32s: \"%s\"", var->prop->name, val==NULL?"(error)":val);
 			}
 		}
-		else if (strncmp(cmd,"set",max(1,strlen(cmd)))==0)
+		else if (strncmp(cmd,"set",fmax(1,strlen(cmd)))==0)
 		{
 			char256 objname;
 			char256 propname;
@@ -1194,7 +1194,7 @@ Retry:
 				 */
 			}
 		}
-		else if (strncmp(cmd,"find ",max(1,strlen(cmd)))==0)
+		else if (strncmp(cmd,"find ",fmax(1,strlen(cmd)))==0)
 		{
 			FINDLIST *fl = NULL;
 			OBJECT *obj = NULL;
@@ -1212,7 +1212,7 @@ Retry:
 			if (exec("gdb --quiet %s --pid=%d",global_execname,global_process_id)<=0)
 				output_debug("unable to start gdb");
 		}
-		else if (strncmp(cmd,"help",max(1,strlen(cmd)))==0)
+		else if (strncmp(cmd,"help",fmax(1,strlen(cmd)))==0)
 		{
 			output_debug("Summary of debug commands\n"
 				"   break             prints all breakpoints\n"

--- a/gldcore/exec.h
+++ b/gldcore/exec.h
@@ -74,10 +74,6 @@ int64 exec_clock(void);
 }
 #endif
 
-#ifndef max
-#define max(n, m) ((n) > (m) ? (n) : (m))
-#endif
-
 #endif
 
 /**@}*/

--- a/gldcore/object.c
+++ b/gldcore/object.c
@@ -1415,7 +1415,7 @@ TIMESTAMP _object_sync(OBJECT *obj, /**< the object to synchronize */
 {
 	CLASS *oclass = obj->oclass;
 	register TIMESTAMP plc_time=TS_NEVER, sync_time;
-	TIMESTAMP effective_valid_to = min(obj->clock+global_skipsafe,obj->valid_to);
+	TIMESTAMP effective_valid_to = fmin(obj->clock+global_skipsafe,obj->valid_to);
 	int autolock = obj->oclass->passconfig&PC_AUTOLOCK;
 
 	/* check skipsafe */

--- a/gldcore/platform.h
+++ b/gldcore/platform.h
@@ -45,8 +45,6 @@
 	#ifndef isfinite
 		#define isfinite finite
 	#endif
-	#define min fmin /**< min macro */
-	#define max fmax /**< max macro */
 	#ifdef X64
 		#define NATIVE int64	/**< native integer size */
 	#else

--- a/gldcore/random.c
+++ b/gldcore/random.c
@@ -782,7 +782,7 @@ double pseudorandom_value(RANDOMTYPE type, /**< the type of distribution desired
 }
 
 /******************************************************************************/
-static double mean(double sample[], unsigned int count)
+static double samp_mean(double sample[], unsigned int count)
 {
 	double sum=0;
 	unsigned int i;
@@ -790,35 +790,31 @@ static double mean(double sample[], unsigned int count)
 		sum += sample[i];
 	return sum/count;
 }
-#ifdef min
-#undef min
-#endif
-static double min(double sample[], unsigned int count)
+
+static double samp_min(double sample[], unsigned int count)
 {
-	double min;
+	double min_val;
 	unsigned int i;
 	for (i=0; i<count; i++)
 	{
-		if (i==0) min=sample[0];
-		else if (sample[i]<min) min=sample[i];
+		if (i==0) min_val=sample[0];
+		else if (sample[i]<min_val) min_val=sample[i];
 	}
-	return min;
+	return min_val;
 }
-#ifdef max
-#undef max
-#endif
-static double max(double sample[], unsigned int count)
+
+static double samp_max(double sample[], unsigned int count)
 {
-	double max;
+	double max_val;
 	unsigned int i;
 	for (i=0; i<count; i++)
 	{
-		if (i==0) max=sample[0];
-		else if (sample[i]>max) max=sample[i];
+		if (i==0) max_val=sample[0];
+		else if (sample[i]>max_val) max_val=sample[i];
 	}
-	return max;
+	return max_val;
 }
-static double stdev(double sample[], unsigned int count)
+static double samp_stdev(double sample[], unsigned int count)
 {
 	double sum=0, mean=0;
 	unsigned int n = 0, i;
@@ -880,10 +876,10 @@ int random_test(void)
 			failed++,output_test("Sample %d is not a finite number!",i--);
 	}
 	errorcount+=report(NULL,0,0,0);
-	errorcount+=report("Mean",mean(sample,count),a,0.01);
-	errorcount+=report("Stdev",stdev(sample,count),0,0.01);
-	errorcount+=report("Min",min(sample,count),a,0.01);
-	errorcount+=report("Max",max(sample,count),a,0.01);
+	errorcount+=report("Mean",samp_mean(sample,count),a,0.01);
+	errorcount+=report("Stdev",samp_stdev(sample,count),0,0.01);
+	errorcount+=report("Min",samp_min(sample,count),a,0.01);
+	errorcount+=report("Max",samp_max(sample,count),a,0.01);
 	if (preverrors==errorcount)	ok++; else failed++;
 	preverrors=errorcount;
 
@@ -898,10 +894,10 @@ int random_test(void)
 			failed++,output_test("Sample %d is not a finite number!",i--);
 	}
 	errorcount+=report(NULL,0,0,0);
-	errorcount+=report("Mean",mean(sample,count),(a+b)/2,0.01);
-	errorcount+=report("Stdev",stdev(sample,count),sqrt((b-a)*(b-a)/12),0.01);
-	errorcount+=report("Min",min(sample,count),a,0.01);
-	errorcount+=report("Max",max(sample,count),b,0.01);
+	errorcount+=report("Mean",samp_mean(sample,count),(a+b)/2,0.01);
+	errorcount+=report("Stdev",samp_stdev(sample,count),sqrt((b-a)*(b-a)/12),0.01);
+	errorcount+=report("Min",samp_min(sample,count),a,0.01);
+	errorcount+=report("Max",samp_max(sample,count),b,0.01);
 	if (preverrors==errorcount)	ok++; else failed++;
 	preverrors=errorcount;
 
@@ -915,10 +911,10 @@ int random_test(void)
 			failed++,output_test("Sample %d is not a finite number!",i--);
 	}
 	errorcount+=report(NULL,0,0,0.01);
-	errorcount+=report("Mean",mean(sample,count),a,0.01);
-	errorcount+=report("Stdev",stdev(sample,count),sqrt(a*(1-a)),0.01);
-	errorcount+=report("Min",min(sample,count),0,0.01);
-	errorcount+=report("Max",max(sample,count),1,0.01);
+	errorcount+=report("Mean",samp_mean(sample,count),a,0.01);
+	errorcount+=report("Stdev",samp_stdev(sample,count),sqrt(a*(1-a)),0.01);
+	errorcount+=report("Min",samp_min(sample,count),0,0.01);
+	errorcount+=report("Max",samp_max(sample,count),1,0.01);
 	if (preverrors==errorcount)	ok++; else failed++;
 	preverrors=errorcount;
 
@@ -933,8 +929,8 @@ int random_test(void)
 			failed++,output_test("Sample %d is not a finite number!",i--);
 	}
 	errorcount+=report(NULL,0,0,0.01);
-	errorcount+=report("Mean",mean(sample,count),a,0.01);
-	errorcount+=report("Stdev",stdev(sample,count),b,0.01);
+	errorcount+=report("Mean",samp_mean(sample,count),a,0.01);
+	errorcount+=report("Stdev",samp_stdev(sample,count),b,0.01);
 	if (preverrors==errorcount)	ok++; else failed++;
 	preverrors=errorcount;
 
@@ -948,9 +944,9 @@ int random_test(void)
 			failed++,output_test("Sample %d is not a finite number!",i--);
 	}
 	errorcount+=report(NULL,0,0,0.01);
-	errorcount+=report("Mean",mean(sample,count),1/a,0.01);
-	errorcount+=report("Stdev",stdev(sample,count),1/a,0.01);
-	errorcount+=report("Min",min(sample,count),0,0.01);
+	errorcount+=report("Mean",samp_mean(sample,count),1/a,0.01);
+	errorcount+=report("Stdev",samp_stdev(sample,count),1/a,0.01);
+	errorcount+=report("Min",samp_min(sample,count),0,0.01);
 	if (preverrors==errorcount)	ok++; else failed++;
 	preverrors=errorcount;
 	
@@ -965,9 +961,9 @@ int random_test(void)
 			failed++,output_test("Sample %d is not a finite number!",i--);
 	}
 	errorcount+=report(NULL,0,0,0.01);
-	errorcount+=report("Mean",mean(sample,count),exp(a+b*b/2),0.01);
-	errorcount+=report("Stdev",stdev(sample,count),sqrt((exp(b*b)-1)*exp(2*a+b*b)),0.1);
-	errorcount+=report("Min",min(sample,count),0,0.1);
+	errorcount+=report("Mean",samp_mean(sample,count),exp(a+b*b/2),0.01);
+	errorcount+=report("Stdev",samp_stdev(sample,count),sqrt((exp(b*b)-1)*exp(2*a+b*b)),0.1);
+	errorcount+=report("Min",samp_min(sample,count),0,0.1);
 	if (preverrors==errorcount)	ok++; else failed++;
 	preverrors=errorcount;
 
@@ -982,9 +978,9 @@ int random_test(void)
 			failed++,output_test("Sample %d is not a finite number!",i--);
 	}
 	errorcount+=report(NULL,0,0,0.01);
-	errorcount+=report("Mean",mean(sample,count),b*a/(b-1),0.01);
-	errorcount+=report("Stdev",stdev(sample,count),sqrt(a*a*b/((b-1)*(b-1)*(b-2))),0.25);
-	errorcount+=report("Min",min(sample,count),a,0.01);
+	errorcount+=report("Mean",samp_mean(sample,count),b*a/(b-1),0.01);
+	errorcount+=report("Stdev",samp_stdev(sample,count),sqrt(a*a*b/((b-1)*(b-1)*(b-2))),0.25);
+	errorcount+=report("Min",samp_min(sample,count),a,0.01);
 	if (preverrors==errorcount)	ok++; else failed++;
 	preverrors=errorcount;
 
@@ -999,8 +995,8 @@ int random_test(void)
 			failed++,output_test("Sample %d is not a finite number!",i--);
 	}
 	errorcount+=report(NULL,0,0,0.01);
-	errorcount+=report("Mean",mean(sample,count),a*sqrt(3.1415926/2),0.01);
-	errorcount+=report("Stdev",stdev(sample,count),sqrt((4-3.1415926)/2*a*a),0.01);
+	errorcount+=report("Mean",samp_mean(sample,count),a*sqrt(3.1415926/2),0.01);
+	errorcount+=report("Stdev",samp_stdev(sample,count),sqrt((4-3.1415926)/2*a*a),0.01);
 	if (preverrors==errorcount)	ok++; else failed++;
 	preverrors=errorcount;
 
@@ -1015,8 +1011,8 @@ int random_test(void)
 			failed++,output_test("Sample %d is not a finite number!",i--);
 	}
 	errorcount+=report(NULL,0,0,0.01);
-	errorcount+=report("Mean",mean(sample,count),a/(a+b),0.01);
-	errorcount+=report("Stdev",stdev(sample,count),sqrt(a*b/((a+b)*(a+b)*(a+b+1))),0.01);
+	errorcount+=report("Mean",samp_mean(sample,count),a/(a+b),0.01);
+	errorcount+=report("Stdev",samp_stdev(sample,count),sqrt(a*b/((a+b)*(a+b)*(a+b+1))),0.01);
 	if (preverrors==errorcount)	ok++; else failed++;
 	preverrors=errorcount;
 
@@ -1031,8 +1027,8 @@ int random_test(void)
 			failed++,output_test("Sample %d is not a finite number!",i--);
 	}
 	errorcount+=report(NULL,0,0,0.01);
-	errorcount+=report("Mean",mean(sample,count),a*b,0.25);
-	errorcount+=report("Stdev",stdev(sample,count),sqrt(a*b*b),0.01);
+	errorcount+=report("Mean",samp_mean(sample,count),a*b,0.25);
+	errorcount+=report("Stdev",samp_stdev(sample,count),sqrt(a*b*b),0.01);
 	if (preverrors==errorcount)	ok++; else failed++;
 	preverrors=errorcount;
 
@@ -1047,8 +1043,8 @@ int random_test(void)
 			output_test("Sample %d is not a finite number!",i--);
 	}
 	errorcount+=report(NULL,0,0,0.01);
-	errorcount+=report("Mean",mean(sample,count),(a+b)/2,0.01);
-	errorcount+=report("Stdev",stdev(sample,count),sqrt((a-b)*(a-b)/24),0.01);
+	errorcount+=report("Mean",samp_mean(sample,count),(a+b)/2,0.01);
+	errorcount+=report("Stdev",samp_stdev(sample,count),sqrt((a-b)*(a-b)/24),0.01);
 	if (preverrors==errorcount)	ok++; else failed++;
 	preverrors=errorcount;
 
@@ -1062,11 +1058,11 @@ int random_test(void)
 			failed++,output_test("Sample %d is not a finite number!",i--);
 	}
 	errorcount+=report(NULL,0,0,0.01);
-	errorcount+=report("Mean",mean(sample,count),4.5,0.01);
-	//report("Stdev",stdev(sample,count),sqrt(9*9/12),0.01);
-	errorcount+=report("Stdev",stdev(sample,count),sqrt(99/12),0.1); /* sqrt((b-a+1)^2-1 / 12)*/ /* 2.87 is more accurate and was Mathematica's answer */
-	errorcount+=report("Min",min(sample,count),0,0.01);
-	errorcount+=report("Max",max(sample,count),9,0.01);
+	errorcount+=report("Mean",samp_mean(sample,count),4.5,0.01);
+	//report("Stdev",samp_stdev(sample,count),sqrt(9*9/12),0.01);
+	errorcount+=report("Stdev",samp_stdev(sample,count),sqrt(99/12),0.1); /* sqrt((b-a+1)^2-1 / 12)*/ /* 2.87 is more accurate and was Mathematica's answer */
+	errorcount+=report("Min",samp_min(sample,count),0,0.01);
+	errorcount+=report("Max",samp_max(sample,count),9,0.01);
 	if (preverrors==errorcount)	ok++; else failed++;
 	preverrors=errorcount;
 

--- a/gldcore/server.c
+++ b/gldcore/server.c
@@ -294,7 +294,7 @@ typedef struct s_httpcnx {
 	char query[1024];
 	char *buffer;
 	size_t len;
-	size_t max;
+	size_t max_size;
 	char *status;
 	char *type;
 	SOCKET s;
@@ -309,8 +309,8 @@ static HTTPCNX *http_create(SOCKET s)
 	HTTPCNX *http = (HTTPCNX*)malloc(sizeof(HTTPCNX));
 	memset(http,0,sizeof(HTTPCNX));
 	http->s = s;
-	http->max = 65536;
-	http->buffer = malloc(http->max);
+	http->max_size = 65536;
+	http->buffer = malloc(http->max_size);
 	return http;
 }
 
@@ -458,19 +458,19 @@ static void http_write(HTTPCNX *http, char *data, size_t len)
 		tmp = (char*)malloc(need*2+1);
 		len = http_rewrite(tmp,data,len,need*2);
 	}
-	if (http->len+len>=http->max)
+	if (http->len+len>=http->max_size)
 	{
 		/* extend buffer */
 		void *old = http->buffer;
-		if (http->len+len < http->max*2)
+		if (http->len+len < http->max_size*2)
 		{
-			http->max *= 2;
+			http->max_size *= 2;
 		}
 		else
 		{
-			http->max = http->len+len+1;
+			http->max_size = http->len+len+1;
 		}
-		http->buffer = malloc(http->max);
+		http->buffer = malloc(http->max_size);
 		memcpy(http->buffer,old,http->len);
 		free(old);
 	}

--- a/gldcore/unit.c
+++ b/gldcore/unit.c
@@ -387,7 +387,7 @@ int unit_derived(char *name,char *derivation)
 				m += pUnit->m;
 				s += pUnit->s;
 				a *= pUnit->a;
-				prec = min(prec,pUnit->prec);
+				prec = fmin(prec,pUnit->prec);
 				break;
 			case '/':
 				c -= pUnit->c;
@@ -397,7 +397,7 @@ int unit_derived(char *name,char *derivation)
 				m -= pUnit->m;
 				s -= pUnit->s;
 				a /= pUnit->a;
-				prec = min(prec, pUnit->prec);
+				prec = fmin(prec, pUnit->prec);
 				break;
 			case '+':
 				b += pUnit->b;
@@ -425,7 +425,7 @@ int unit_derived(char *name,char *derivation)
 							m += lastUnit->m * repeat;
 							s += lastUnit->s * repeat;
 							a *= pow(lastUnit->a, repeat);
-							prec = min(prec, lastUnit->prec);
+							prec = fmin(prec, lastUnit->prec);
 							break;
 						case '/':
 							c -= lastUnit->c * repeat;
@@ -435,7 +435,7 @@ int unit_derived(char *name,char *derivation)
 							m -= lastUnit->m * repeat;
 							s -= lastUnit->s * repeat;
 							a /= pow(lastUnit->a, repeat);
-							prec = min(prec, lastUnit->prec);
+							prec = fmin(prec, lastUnit->prec);
 							break;
 						default:
 							throw_exception("%s(%d): ^ not allowed after '%c' at '%s'", filepath, linenum, lastOp, term);

--- a/tape/histogram.cpp
+++ b/tape/histogram.cpp
@@ -48,8 +48,8 @@ histogram::histogram(MODULE *mod)
 			PT_char1024, "group", PADDR(group),PT_DESCRIPTION,"the GridLAB-D group expression to use for this histogram",
 			PT_char1024, "bins", PADDR(bins),PT_DESCRIPTION,"the specific bin values to use",
 			PT_char256, "property", PADDR(property),PT_DESCRIPTION,"the property to sample",
-			PT_double, "min", PADDR(min),PT_DESCRIPTION,"the minimum value of the auto-sized bins to use",
-			PT_double, "max", PADDR(max),PT_DESCRIPTION,"the maximum value of the auto-sized bins to use",
+			PT_double, "min", PADDR(min_val),PT_DESCRIPTION,"the minimum value of the auto-sized bins to use",
+			PT_double, "max", PADDR(max_val),PT_DESCRIPTION,"the maximum value of the auto-sized bins to use",
 			PT_double, "samplerate[s]", PADDR(sampling_interval),PT_DESCRIPTION,"the rate at which samples are read",
 			PT_double, "countrate[s]", PADDR(counting_interval),PT_DESCRIPTION,"the reate at which bins are counted and written",
 			PT_int32, "bin_count", PADDR(bin_count),PT_DESCRIPTION,"the number of auto-sized bins to use",
@@ -60,8 +60,8 @@ histogram::histogram(MODULE *mod)
 		memset(group, 0, 1025);
 		memset(bins, 0, 1025);
 		memset(property, 0, 33);
-		min = 0;
-		max = -1;
+		min_val = 0;
+		max_val = -1;
 		bin_count = -1;
 		sampling_interval = -1.0;
 		counting_interval = -1.0;
@@ -282,10 +282,10 @@ int histogram::init(OBJECT *parent)
 	/*
 	 *	This will create a uniform partition over the specified range
 	 */
-	if((bin_count > 0) && (min < max))
+	if((bin_count > 0) && (min_val < max_val))
 	{
 		int i=0;
-		double range = max - min;
+		double range = max_val - min_val;
 		double step = range/bin_count;
 		//throw("Histogram bin_count is temporarily disabled.");
 		bin_list = (BIN *)gl_malloc(sizeof(BIN) * bin_count);
@@ -295,7 +295,7 @@ int histogram::init(OBJECT *parent)
 		}
 		memset(bin_list, 0, sizeof(BIN) * bin_count);
 		for(i = 0; i < bin_count; i++){
-			bin_list[i].low_val = min + i * step;
+			bin_list[i].low_val = min_val + i * step;
 			bin_list[i].high_val = bin_list[i].low_val + step;
 			bin_list[i].low_inc = 1;
 			bin_list[i].high_inc = 0;

--- a/tape/histogram.h
+++ b/tape/histogram.h
@@ -44,8 +44,8 @@ public:
 	char256 property;
 	char1024 bins;
 	int32 bin_count;
-	double min;
-	double max;
+	double min_val;
+	double max_val;
 	double sampling_interval;	//	add values into bins
 	double counting_interval;	//	dump the bins into a row
 	int32 limit;			//	number of lines to write

--- a/tape/memory.c
+++ b/tape/memory.c
@@ -50,7 +50,7 @@ char *memory_read_player(struct player *my,char *buffer,unsigned int size)
 		double *ptr = (double*)my->memory->buffer->prop->addr + my->memory->index;
 		my->memory->index += 2;
 		sprintf(temp,"%"FMT_INT64"d,%lg",(TIMESTAMP)ptr[0],ptr[1]);
-		strncpy(buffer,temp,min(size,sizeof(temp)));
+		strncpy(buffer,temp,fmin(size,sizeof(temp)));
 		return buffer;
 	}
 }
@@ -178,7 +178,7 @@ int memory_open_shaper(struct shaper *my, char *fname, char *flags)
 	while (fgets(line,sizeof(line),my->fp)!=NULL)
 	{
 		unsigned char *hours, *days, *months, *weekdays;
-		char min[256],hour[256],day[256],month[256],weekday[256],value[32];
+		char minute[256],hour[256],day[256],month[256],weekday[256],value[32];
 		char *p=line;
 		linenum++;
 		while (isspace(*p)) p++;
@@ -186,13 +186,13 @@ int memory_open_shaper(struct shaper *my, char *fname, char *flags)
 		if (strcmp(group,"")!=0 && (isdigit(p[0]) || p[0]=='*'))
 		{	/* shape value */
 			int h, d, m, w;
-			if (sscanf(line,"%s %s %s %s %[^,],%[^,\n]",min,hour,day,month,weekday,value)<6)
+			if (sscanf(line,"%s %s %s %s %[^,],%[^,\n]",minute,hour,day,month,weekday,value)<6)
 			{
 				gl_error("%s(%d) : shape '%s' has specification '%s'", memory, linenum, group, line);
 				continue;
 			}
 			/* minutes are ignored right now */
-			if (min[0]!='*') gl_warning("%s(%d) : minutes are ignored in '%s'", memory, linenum, line);
+			if (minute[0]!='*') gl_warning("%s(%d) : minutes are ignored in '%s'", memory, linenum, line);
 			hours=hourmap(hour);
 			days=daymap(day);
 			months=monthmap(month);

--- a/tape/metrics_collector.cpp
+++ b/tape/metrics_collector.cpp
@@ -980,7 +980,7 @@ int metrics_collector::write_line(TIMESTAMP t1, OBJECT *obj){
 		curr_index = 1;
 	}
 
-	next_write = min(next_write + interval_length, gl_globalstoptime);
+	next_write = std::min(next_write + interval_length, gl_globalstoptime);
 	first_write = false;
 
 	return 1;

--- a/tape/metrics_collector_writer.cpp
+++ b/tape/metrics_collector_writer.cpp
@@ -293,7 +293,7 @@ TIMESTAMP metrics_collector_writer::postsync(TIMESTAMP t0, TIMESTAMP t1){
 	if(next_write <= t1){
 		interval_write = true;
 		last_write = t1;
-		next_write = min(t1 + interval_length, final_write);
+		next_write = std::min(t1 + interval_length, final_write);
 	}
 
 	// the interval recorders have already returned t1+interval_length, earlier in the sequence.

